tests/functional/aws-node-sdk/test/bucket/aclUsingPredefinedGroups.js

```diff
@@ -0,0 +1,634 @@
+const assert = require('assert');
+const AWS = require('aws-sdk');
+const { errors } = require('arsenal');
+
+const withV4 = require('../support/withV4');
+const BucketUtility = require('../../lib/utility/bucket-util');
+const constants = require('../../../../../constants');
+const { VALIDATE_CREDENTIALS, SIGN } = AWS.EventListeners.Core;
+
+withV4(sigCfg => {
+    const ownerAccountBucketUtil = new BucketUtility('default', sigCfg);
+    const otherAccountBucketUtil = new BucketUtility('lisa', sigCfg);
+    const s3 = ownerAccountBucketUtil.s3;
+
+    const testBucket = 'predefined-groups-bucket';
+    const testKey = '0.txt';
+    const ownerObjKey = 'account.txt';
+    const testBody = '000';
+
+    function unauthenticatedRequest(operation, params, callback) {
+        const bucketUtil = new BucketUtility('default', sigCfg);
+        const request = bucketUtil.s3[operation](params);
+        request.removeListener('validate', VALIDATE_CREDENTIALS);
+        request.removeListener('sign', SIGN);
+        request.send(callback);
+    }
+
+    // tests for authenticated user(signed) and anonymous user(unsigned)
+    [true, false].forEach(auth => {
```

Comment:
splitting all these tests with a bool here is making it quite hard to read and adds at least one if check to every test, I think it would be better to create two describes instead.

Reply:
The tests are the same for both the cases and that is why I had to do this. I originally had separate tests.

Reply:
I would not consider the tests the same if you need an if/else in the test based on the condition. It is not like a table test where you are testing many inputs against the same method and output. `auth` is used as a conditional check a lot.

Reply:
Refactored a bit.

Reply:
I will accept it, but I tend to dislike conditionals in tests as I think a straight top-down given-when-then flow works more consistently.